lib/_stream_readable.js

```diff
@@ -458,6 +458,12 @@ Readable.prototype.read = function(n) {
       endReadable(this);
   }
 
+  // Drop data if '_dumped' flag exists. This is for http.IncomingMessage.
+  // When http.ClientRequest.abort() is called, the '_dumped' flag of
+  // http.IncomingMessage is set to 'true'.  Then http.ClientRequest
+  // shouldn't receive any 'data' event.
+  if (this._dumped === true) ret = null;
```

Comment:
I'm really -1 with this change, as it introduces more coupling between http and streams. Can you move this in http.IncomingMessage? If it is not possible, we should think about adding some public facing API to support this usecase. Where is this._dumped written?